File: AppSets-Android/appsets/src/main/kotlin/xcj/app/appsets/worker/LocalSyncWorker.kt
package xcj.app.appsets.worker

import android.content.Context
import androidx.work.CoroutineWorker
import androidx.work.WorkerParameters
import xcj.app.appsets.usecase.RelationsUseCase
import xcj.app.starter.android.util.PurpleLogger

class LocalSyncWorker(
    context: Context,
    params: WorkerParameters
) : CoroutineWorker(context, params) {

    companion object {
        private const val TAG = "LocalSyncWorker"
    }

    override suspend fun doWork(): Result {
        PurpleLogger.current.d(TAG, "doWork")
        RelationsUseCase.getInstance().initRelationFromLocalDB()
        return Result.success()
    }
}

File: AppSets-iOS/AppSets/com/app/appsets/worker/LocalSyncWorker.swift
//
//  LocalSyncWorker.swift
//  AppSets
//
//  Created by caiju Xu on 2024/7/12.
//

import Foundation

class LocalSyncWorker : ListenableWorker {
    
    private static let TAG = "LocalSyncWorker"
    
    override func doWork() async -> Result {
        PurpleLogger.current.d(LocalSyncWorker.TAG, "doWork")
        await MainActor.run{
            RelationsUseCase.Instance.initRelationFromLocalDB()
        }
        
        return Result.Success
    }
    
}
